tests/utils_tests/test_numberformat.py

```diff
@@ -56,3 +56,41 @@ def test_decimal_numbers(self):
         self.assertEqual(nformat(Decimal('1234'), '.', grouping=2, thousand_sep=',', force_grouping=True), '12,34')
         self.assertEqual(nformat(Decimal('-1234.33'), '.', decimal_pos=1), '-1234.3')
         self.assertEqual(nformat(Decimal('0.00000001'), '.', decimal_pos=8), '0.00000001')
+
+    def test_custom_decimal_numbers(self):
+        self.assertEqual(nformat(Money('1234'), '.'), '$1,234.00')
+        self.assertEqual(nformat(Money('1234.2'), '.'), '$1,234.20')
+        self.assertEqual(nformat(Money('1234'), '.', decimal_pos=2), '$1,234.00')
+        self.assertEqual(nformat(Money('1234'), '.', grouping=2, thousand_sep=','), '$1,234.00')
+        self.assertEqual(nformat(Money('1234'), '.', grouping=2, thousand_sep=',', force_grouping=True), '$1,,2,34.00')
```

Comment:
The lower one disappears because the custom __str__ rounds to the nearest cent, but yeah this ones's bad.